README.md

```diff
@@ -8,77 +8,65 @@ npm install hypercore-protocol
 
 [![build status](https://travis-ci.org/mafintosh/hypercore-protocol.svg?branch=master)](https://travis-ci.org/mafintosh/hypercore-protocol)
 
+For detailed info on the messages sent on each channel see [simple-hypercore-protocol](https://github.com/mafintosh/simple-hypercore-protocol)
+
 ## Usage
 
 ``` js
-var protocol = require('hypercore-protocol')
+const Protocol = require('hypercore-protocol')
 
 // create two streams with hypercore protocol
-var streamA = protocol({ id: 'a' })
-var streamB = protocol({ id: 'b' })
+const streamA = new Protocol(true)
+const streamB = new Protocol(false)
 
 // open two feeds specified by a 32 byte key
-var key = Buffer.from('deadbeefdeadbeefdeadbeefdeadbeef')
-var feed = streamA.feed(key)
-var remoteFeed = streamB.feed(key)
+const key = Buffer.from('deadbeefdeadbeefdeadbeefdeadbeef')
+const feed = streamA.open(key)
+const remoteFeed = streamB.open(key, {
+  // listen for data in remote feed
+  ondata (message) {
+    console.log(message.value.toString())
+  }
+})
 
 // add data to feed
 feed.data({ index: 1, value: '{ block: 42 }'})
 
-// listen data in remoteFeed
-remoteFeed.on('data', function (message) {
-  console.log(message.value.toString())
-})
-
 streamA.pipe(streamB).pipe(streamA)
 ```
 
 `output => { block: 42 }`
 
 ## API
 
-#### `var stream = protocol([options])`
+#### `const stream = new Protocol(initiator, [options])`
 
 Create a new protocol duplex stream.
 
 Options include:
 
 ``` js
 {
-  id: optionalPeerId, // you can use this to detect if you connect to yourself
-  live: keepStreamOpen, // signal to the other peer that you want to keep this stream open forever
```

Comment:
Live is the new default and only mode?
Which means that proto.expectedFeeds is nolonger an [issue?](https://github.com/mafintosh/hypercore/issues/219)

Reply:
Live is gone from the replication protocol, but will still be an option when you replicate a data structure.

Yep, want to see if I can come with a better option than expected feeds

Reply:
Very happy to see this! :+1:

Reply:
We've added the `.seal()` api instead.

Reply:
I dig the seal api

Reply:
Hmm.. wait this means that only the topmost replication session can safely call seal()...
If i have a single protocol-stream that is shared among a heirarchy of cores:

```
main stream -v
             | - hypercore A
             | - hypercore B
             |
             | - hyperdrive v9 -v
             |                  | - meta hypercore
             |                  | - content hypercore
             |
             |
             | - corestore -v   // <-- calls seal() when core [1,2,3] is done.
                            | - core 1
                            | - core 2
                            | - core 3
```
In the configuration given above, the core store has no way of telling if there are other feeds present on the stream nor is there a way to detect if the stream belongs to itself.

So unless i've misunderstood the api, then if the corestore calls  `seal()` it will seal the entire stream for all other neighbours?

Reply:
we updated this with the prefinalize api